evennia/objects/objects.py

```diff
@@ -959,7 +959,22 @@ def find_clone_key():
                       if obj.key.startswith(key) and obj.key.lstrip(key).isdigit())
             return "%s%03i" % (key, num)
         new_key = new_key or find_clone_key()
-        return ObjectDB.objects.copy_object(self, new_key=new_key)
+        new_obj = ObjectDB.objects.copy_object(self, new_key=new_key)
```

Comment:
Hm. It's not strictly part of this PR, but `ObjectDB.objects.copy_object` takes a lot of `**kwargs` for modifying the resulting copy. I see no reason why the `copy` method could not take `**kwargs` too (they are optional, so would not affect current functionality) and pass them on to `copy_object`. Maybe those `**kwargs` could then also be passed into the new hook.